### PR TITLE
Take into account @Produces annotations from interfaces and superclasses in JaxrsEndPointValidationInterceptor

### LINKED ISSUE #29253: Resteasy + Hibernate Validator : @Produces on super interface is not taken into account, JSON is default
### Describe the bug

I have a `ResourceInterface` and a `ResourceImplementation`.

The `ResourceInterface` defines everything related to JAX-RS and OpenAPI.x

It looks like this

```
// ...
@Produces("application/xml")
@Consumes("application/xml")
public interface ResourceInterface {
    //...
    @POST
    Output method(@Valid Input input);
    //....
```

The `ResourceImplementation` looks like this

```
@ApplicationScoped
public class ResourceImplementation implements ResourceInterface {
    //...
    public Output method(Input input) {
        //....
    }
}
```

When I get a validation exception, it gets outputted as JSON, even though I specified XML on the class as described by the JAX-RS specification.

The guilty piece of code is the following, that only checks on the invoked method and class but not the super.

This comes from `JaxrsEndPointValidationInterceptor`

```
    @AroundInvoke
    @Override
    public Object validateMethodInvocation(InvocationContext ctx) throws Exception {
        try {
            return super.validateMethodInvocation(ctx);
        } catch (ConstraintViolationException e) {
            throw new ResteasyViolationExceptionImpl(e.getConstraintViolations(), getProduces(ctx.getMethod()));
        }
    }

    @AroundConstruct
    @Override
    public void validateConstructorInvocation(InvocationContext ctx) throws Exception {
        super.validateConstructorInvocation(ctx);
    }

    private List<MediaType> getProduces(Method method) {
        MediaType[] producedMediaTypes = MediaTypeHelper.getProduces(method.getDeclaringClass(), method);

        if (producedMediaTypes == null) {
            if (resteasyConfigSupport.isJsonDefault()) {
                return JSON_MEDIA_TYPE_LIST;
            }

            return Collections.emptyList();
        }

        return Arrays.asList(producedMediaTypes);
    }

```

### Expected behavior

Return validation error as XML

### Actual behavior

Returns validation error as JSON

PS : Of course, a workaround is possible. The annotation should be put on the implementating class' method. It doesn't work when put on the interface's class method.

### How to Reproduce?

Create the classes as described above and produce a validation error.
It will get outputed as JSON and not XML.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gsmet**
Note that we are not following strictly the JAX-RS specification but we try to analyze the @Produces annotations on a best effort basis. Hopefully, it will be good enough.

Fixes #29253

### CODE REVIEW

**Review by yrodiere — APPROVED** [review]
LGTM, I only have a doubt on the performance impact, see below. I'll let you judge whether a cache is necessary or overkill.

File: extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java [hunk 1]
```diff
@@ -44,17 +49,75 @@ public void validateConstructorInvocation(InvocationContext ctx) throws Exceptio
...
+        if (currentClass.equals(originalMethod.getDeclaringClass())) {
+            return getProduces(originalMethod.getAnnotation(Produces.class));
+        }
+
+        try {
+            return getProduces(currentClass
+                    .getMethod(originalMethod.getName(), originalMethod.getParameterTypes()).getAnnotation(Produces.class));
+        } catch (NoSuchMethodException | SecurityException e) {
+            // we don't have a visible method around, let's ignore this class
             return Collections.emptyList();
```
[the remark this answers is not in the document]
Comment (gsmet) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java:107]: We could use a CHM but I wouldn't make Caffeine a required dependency.

[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java:107]: Ah, I thought it was a dependency in core. CHM makes sense then.

[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java:107]: @gsmet Do you plan to add that cache, or should I just merge this PR as is?

[the remark this answers is not in the document]
Comment (gsmet) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java]: @yrodiere could you have a look at the update?

**Review by yrodiere — APPROVED** [review]

File: extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yrodiere) [reply extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/JaxrsEndPointValidationInterceptor.java]: LGTM, thanks

**Comment by yrodiere** [comment]
The failure looks unrelated. Merging.

**Comment by YassinHajaj** [comment]
Nice, thank you guys, hope my little project was helpful ;)

**Pull request merged**